### ISSUE TITLE
segmentation fault ubuntu 16.04

### ISSUE THREAD

**Issue by wli3** [dotnet/runtime#460]
_From @ceka on Monday, November 25, 2019 12:57:39 PM_

i've got strange results after upgrading to dotnet-3.0 sdk.

1) dotnet run never finish
2) dotnet build works as usual
3) dotnet run --no-build runs after step 2)
4) dotnet run --no-build sometimes DEFUNCT
5) dotnet run --no-build sometimes goes into segmentation fault

tested on new console project.
```bash
ck@poznan:~/src/test2$ ls
Program.cs  test2.csproj
```

_Copied from original issue: dotnet/sdk#3965_

**Comment by wli3** [comment]
_From @ceka on Monday, November 25, 2019 12:58:35 PM_

```bash
ck@poznan:~$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.101
 Commit:    bc5f8df0f5

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.0.101/

Host (useful for support):
  Version: 3.0.1
  Commit:  19942e7199

.NET Core SDKs installed:
  2.1.4 [/usr/share/dotnet/sdk]
  2.1.802 [/usr/share/dotnet/sdk]
  2.2.402 [/usr/share/dotnet/sdk]
  3.0.101 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.14 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.14 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.14 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by wli3** [comment]
_From @ceka on Monday, November 25, 2019 12:59:05 PM_

```bash
ck@poznan:~$ dotnet --version
3.0.101
```

**Comment by wli3** [comment]
_From @ceka on Monday, November 25, 2019 1:03:32 PM_

also ^C never finish:

```bash
ck@poznan:~/src/test2$ dotnet run
^C
Attempting to cancel the build...
```

**Comment by ceka** [comment]
@wli3 Please let me know if You need additional information.

**Comment by jkotas** [comment]
Could you please share your test project? (test2.csproj + Program.cs)

**Comment by ceka** [comment]
> 
> 
> Could you please share your test project? (test2.csproj + Program.cs)

this is new empty console project from dotnet new console command
```cs
using System;

namespace test2
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }
    }
}
```

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>

</Project>
```

**Comment by wfurt** [comment]
I was not ab le to reproduce this. Is this container or full OS and do you have SELinux enabled? You can try to run what ever is failing under `strace -f -s 200 - t XXX`. Did you get dotnet via package feed or did you get just binary? Do you have any custom build package .. like openssl? Can you please share yout environmental variables?

**Comment by ceka** [comment]
vps/openvz
Linux poznan 2.6.32-042stab127.2 #1 SMP Thu Jan 4 16:41:44 MSK 2018 x86_64 x86_64 x86_64 GNU/Linux

package feed
no selinux
standalone openssl
OpenSSL 1.0.2g  1 Mar 2016

[strace.gz](https://github.com/dotnet/runtime/files/3950158/strace.gz)

**Comment by janvorli** [comment]
I wonder if it is yet another issue specific to the openvz. There was a recent fix to a CPU count issue on openvz @ceka, could you please try the latest alpha version of .NET Core 5.0 to see if the issue is still there? You can get it here:
https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-linux-x64.tar.gz. Just untar the file and use dotnet from the root folder of the package. You'll also need to change the TargetFramework in your .csproj  to netcoreapp5.0.

**Comment by ceka** [comment]
@janvorli  Thx, that version worked.

**Comment by jkotas** [comment]
@ceka Thanks for the confirmation

**Comment by ceka** [comment]
> 
> 
> @ceka Thanks for the confirmation

will there be a planed update for dotnet3 or i should use version alpha 5.0 or downgrade to 2.x ?

**Comment by jkotas** [comment]
This fix is scheduled to ship in .NET Core 3.1.2 update next year.

**Comment by ceka** [comment]
I really appreciate Your help, Thank You!
